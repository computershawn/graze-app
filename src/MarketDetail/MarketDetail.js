import React, { Component } from 'react'
import Nav from '../Nav/Nav'
import './MarketDetail.css'
import dummyStore from '../dummy-store'

class MarketDetail extends Component {
  render() {
    let urlMarketId = this.props.match.params.marketId
    let marketItem = dummyStore.markets.find(market => (market.id.toString() === urlMarketId.toString()))
    let content = (
      <>
        <Nav />
        <main>
          <section>
            <header></header>
            <div>No market with id {urlMarketId}</div>
          </section>
        </main>
      </>
    )
    if (marketItem) {
      const { title, /*hero_image,*/ schedule, address, categories, vendor_list, description } = marketItem
      const paragraphs = description.split('****')
      const descriptionHTML = paragraphs.map((para, i) => <p key={i}>{para}</p>)
      const cats = categories.join(', ')
      const vendorList = vendor_list.map(vendor => {
        return <li key={vendor.id}>{vendor.title} (Stall {vendor.stall})</li>
      });

      content = (
        <>
          <Nav />
          <main>
            <header className="hero" role="banner"></header>
            <section>
              <header>
                <h2>{title}</h2>
              </header>
              <div className="description">
                <p><span className="detail-label">When:</span> {schedule}</p>
                <p><span className="detail-label">Where:</span> {address} | <a href="https://maps.google.com">Map</a></p>
                <p><span className="detail-label">Categories:</span> {cats}</p>
              </div>
            </section>
            <section>
              <header>
                <h3>Information</h3>
              </header>
              {descriptionHTML}
              <ul className="vendor-list">
                {vendorList}
              </ul>
            </section>
          </main>
        </>
      )
    }

    return content
  }
}


export default MarketDetail;