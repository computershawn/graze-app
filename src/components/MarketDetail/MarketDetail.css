.hero {
  /* background-image: url('https://via.placeholder.com/960x540.png?text=hero+image'); */
  background-image: url('https://images.unsplash.com/photo-1498579397066-22750a3cb424?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=2850&q=80');
  background-repeat: none;
  background-position: center;
  background-size: cover;
}

.description p {
  margin-left: 6px;
  padding: 6px 15px;
}

.vendor-list li {
  text-transform: capitalize
}

.detail-label {
  font-weight: 600;
}