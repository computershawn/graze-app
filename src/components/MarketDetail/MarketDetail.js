import React, { Component } from 'react'
import Nav from '../Nav/Nav'
import './MarketDetail.css'
import MarketDataContext from '../../MarketDataContext'

class MarketDetail extends Component {
  static contextType = MarketDataContext;
  render() {
    const { markets=[], vendors=[] } = this.context;
    let urlMarketId = this.props.match.params.marketId
    let marketItem = markets.find(market => (market.id.toString() === urlMarketId.toString()))
    let content = (
      <>
        <Nav />
        <main>
          <section>
            <header></header>
            <div>No market with id {urlMarketId}</div>
          </section>
        </main>
      </>
    )
    if (marketItem) {
      //const { market_name, hero_image, schedule, market_location, categories, market_description } = marketItem
      const { market_name, schedule, market_location, market_description } = marketItem
      const paragraphs = market_description.split('****')
      const descriptionHTML = paragraphs.map((para, i) => <p key={i}>{para}</p>)
      const marketVendors = vendors.filter(vendor => {
        return vendor.market_id.toString() === urlMarketId.toString()
      })
      const vendorListHTML = marketVendors.map(vendor => {
        return <li key={vendor.id}>{vendor.vendor_name} (Stall {vendor.market_stall})</li>
      });

      content = (
        <>
          <Nav />
          <main>
            <header className="hero" role="banner"></header>
            <section>
              <header>
                <h2>{market_name}</h2>
              </header>
              <div className="description">
                <p><span className="detail-label">When:</span> {schedule}</p>
                <p><span className="detail-label">Where:</span> {market_location} | <a href="https://maps.google.com">Map</a></p>
                {/* <p><span className="detail-label">Categories:</span> {cats}</p> */}
              </div>
            </section>
            <section>
              <header>
                <h3>About This Market</h3>
              </header>
              {descriptionHTML}
              <ul className="vendor-list">
                {vendorListHTML}
              </ul>
            </section>
          </main>
        </>
      )
    }

    return content
  }
}


export default MarketDetail;