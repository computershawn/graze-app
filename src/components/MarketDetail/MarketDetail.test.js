import React from 'react'
import ReactDOM from 'react-dom'
import { BrowserRouter } from 'react-router-dom'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import MarketDetail from './MarketDetail'

describe(`MarketDetail component`, () => {
  const p = {
    match: {
      params: {
        marketId: 1
      }
    }
  }
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <BrowserRouter>
        <MarketDetail {...p} />
      </BrowserRouter>,
      div
    );
  })
  it('Renders quite nicely', () => {
    const wrapper = shallow(<MarketDetail {...p} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})


