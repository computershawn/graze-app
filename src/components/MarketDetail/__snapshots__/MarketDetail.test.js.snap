// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarketDetail component Renders quite nicely 1`] = `
<Fragment>
  <Nav />
  <main
    className="market-detail-page"
  >
    <section>
      <header />
      <div>
        No market with id 
        1
      </div>
    </section>
  </main>
</Fragment>
`;
